emily/cdk/lib/emily-stack.ts

```diff
@@ -115,50 +147,55 @@ export class EmilyStack extends cdk.Stack {
             nonKeyAttributes: [
                 "RequestId",
                 "StacksBlockHash",
+                "StacksBlockHeight",
                 "Recipient",
                 "Amount",
+                "LastUpdateBlockHash",
             ]
         });
 
-        // TODO: Add an additional GSI for querying by user; not required for MVP.
+        // TODO(TBD): Add an additional GSI for querying by user; not required for MVP.
         return table;
     }
 
     /**
      * Creates or updates a DynamoDB table for chain state.
-     * @param {EmilyStackProps} props The stack properties.
+     * @param {string} tableId The id of the table AWS resource.
+     * @param {string} tableName The name of the DynamoDB table.
      * @returns {dynamodb.Table} The created or updated DynamoDB table.
      * @post A DynamoDB table is returned without additional configuration.
      */
-    createOrUpdateChainstateTable(props: EmilyStackProps): dynamodb.Table {
+    createOrUpdateChainstateTable(
+        tableId: string,
+        tableName: string,
+    ): dynamodb.Table {
         // Create DynamoDB table to store the messages. Encrypted by default.
-        const tableId: string = 'ChainstateTable';
         return new dynamodb.Table(this, tableId, {
-            tableName: EmilyStackUtils.getResourceName(tableId, props),
+            tableName: tableName,
             partitionKey: {
-                name: 'BlockHeight',
+                name: 'Height',
                 type: dynamodb.AttributeType.NUMBER,
             },
             sortKey: {
-                name: 'BlockHash',
-                type: dynamodb.AttributeType.BINARY,
+                name: 'Hash',
```

Comment:
Thought about this more and realized that my views are slightly more complicated than I may have implied. I think we should follow the conventions of the language for things like column or table names, or maybe some common project/organizational convention.

It looks like the vast majority of AWS examples use `PascalCase` for attribute and table names, which isn't surprising, but other sources that I've seen (either open source or closed source) follow the convention of the language. And I think we should do the same.

---

One other thing that we should consider is using `pk` and `sk` for the partition and sort key names. I think here the names we have make sense, but I can see someone being able to make a compelling case that we should use that convention here as well (I haven't internalized enough of the dynamoDB design practices to make a compelling case why). So not pushing for it, just wanted to put it on our radar if it wasn't already.

Reply:
For table data I try to consider the information in the table on its own. It's a chainstate table and each entry corresponds to a block; calling each entry a `BlockHeight` and `BlockHash` is redundant, but `Height` and `Hash` are not. In my head it's like making a struct `Person` and then giving the its parameters names like `PersonName` and `PersonHeight`.

I also believe in separation of implementation and data, and this does go directly against what you're suggesting: I think the database should have its data structured and named explicitly for the idioms associated with the database, and that the language underneath should be considered a wholly separate layer that interfaces between the idiomatically structured database and the idiomatically structured Api calls.

This is why I've, for the most part, redefined the objects in the database and the responses because I believe they should be structurally decoupled like false cognates. Doing this clarifies what changes impact different parts of the code base and clearly separates different components so there aren't any unexpected side effects.

Reply:
> For table data I try to consider the information in the table on its own. It's a chainstate table and each entry corresponds to a block; calling each entry a `BlockHeight` and `BlockHash` is redundant, but `Height` and `Hash` are not.

Yeah, I agree, that makes perfect sense. The only thing I was concerned about was ambiguity. For stacks blocks there are two relevant hashes: the sortition consensus hash; and the block ID, which is a hash of the header and the consensus hash. I wanted to avoid any confusion between the two (but maybe it's unlikely that there is any confusion between the two?). So maybe we should call it `block_id` (or `BlockId` or whatever).

> I also believe in separation of implementation and data, and this does go directly against what you're suggesting: I think the database should have its data structured and named explicitly for the idioms associated with the database[.]

I also agree here too. It's just not clear that attribute name casing is one the idioms of the DynamoDB (well, maybe it's just not clear to me). What I've seen is that people often use their organization style guild or their programming language style for that decision for DynamoDB. Also, the reason why I think we should consider renaming the partition and sort keys to `pk` and `sk` is because of a naming idiom for DynamoDB. But this table has only two attributes so maybe we should probably skip following that idiom.

---

Unrelated to this PR, why do we have a `chainstate` table and endpoints? We could just tell users to use the Hiro API for that info. I'll put this in a ticket or something.

Reply:
> Unrelated to this PR, why do we have a `chainstate` table and endpoints? We could just tell users to use the Hiro API for that info. I'll put this in a ticket or something.

I've open https://github.com/stacks-network/sbtc/discussions/321 for this.

As for the casing stuff, my last points were to add clarification on where I was coming from (not to double down or reiterate). I'm okay with whatever you decide here.

Reply:
Ah, I would not be surprised if public idioms are different from Amazon idioms, but I'd rather default to Amazon Idioms for now because that's what the code currently does. We can change this implementation detail later.